browser/src/layer/ContentControlLayer.js

```diff
@@ -0,0 +1,228 @@
+		}
+		else if (this._json.action === 'hide') {
+			this.onRemove();
+			this.map.off('zoomstart', this._onZoomStart, this);
+			this.map.off('zoomend', this._onZoomEnd, this);
+		}
+	},
+
+	_drawPolygon: function () {
+		var rectArray = L.Bounds.parseArray(this._json.rectangles);
+		var rectangles = rectArray.map(function (rect) {
+			return rect.getPointArray();
+		});
+
+		var docLayer = this.map._docLayer;
+		this._pointSet = CPolyUtil.rectanglesToPointSet(rectangles,
+			function (twipsPoint) {
+				var corePxPt = docLayer._twipsToCorePixels(twipsPoint);
+				corePxPt.round();
+				return corePxPt;
+			});
+
+		if (!this._frame) {
+			this._frame = new CPolygon(this._pointSet, this._polyAttri);
+			this.map._docLayer._canvasOverlay.initPath(this._frame);
+		}
+		this._frame.setPointSet(this._pointSet);
+	},
+
+	_addDropDownBtn: function() {
+		var matches = this._json.rectangles.match(/\d+/g);
+
+		//consider first rectangle to position dropdownbutton
+		var rectangle = [parseInt(matches[0]), parseInt(matches[1]), parseInt(matches[2]), parseInt(matches[3])];
+
+		var topLeftTwips = new L.Point(rectangle[0], rectangle[1]);
+		var offset = new L.Point(rectangle[2], rectangle[3]);
+		var bottomRightTwips = topLeftTwips.add(offset);
+		var buttonAreaTwips = [topLeftTwips, bottomRightTwips];
+
+		var frameArea = new L.Bounds(
+			this.map._docLayer._twipsToPixels(topLeftTwips),
+			this.map._docLayer._twipsToPixels(bottomRightTwips));
+
+		var size = frameArea.getSize();
+		var origin = this.map.getPixelOrigin();
+		var panePos = this.map._getMapPanePos();
+		this._framePos = new L.Point(Math.round(frameArea.min.x + panePos.x - origin.x), Math.round(frameArea.min.y + panePos.y - origin.y));
+		this._frameWidth = Math.round(size.x);
+		this._frameHeight = Math.round(size.y);
+
+		var icon =  L.divIcon({
+			className: 'writer-drop-down-marker',
+			iconSize: [this._frameHeight, this._frameHeight],
+			iconAnchor: [0, 0]
+		});
+
+		this._dropdownButton.setIcon(icon);
+
+		// Then convert to unit which can be used on the layer.
+		var buttonAreaLatLng = new L.LatLngBounds(
+			this.map._docLayer._twipsToLatLng(buttonAreaTwips[0], this.map.getZoom()),
+			this.map._docLayer._twipsToLatLng(buttonAreaTwips[1], this.map.getZoom()));
+
+		var pos = buttonAreaLatLng.getNorthEast();
+		this._dropdownButton.setLatLng(pos);
+		this.map.addLayer(this._dropdownButton);
+	},
+
+	_onClickDropdown: function(event) {
+		this.map.fire('jsdialog', {data: this._openDropdownJson(), callback: this._callback});
+		L.DomEvent.stopPropagation(event);
+	},
+
+	_openDropdownJson: function() {
+		if (!this._json.items)
+			return;
+
+		var json = {
+			'children': [
+				{
+					'id': 'container',
+					'type': 'container',
+					'text': '',
+					'enabled': true,
+					'children': [
+						{
+							'id': 'dropdownlist',
+							'type': 'treelistbox',
+							'text': '',
+							'enabled': true,
+							'children': [
+								{
+									'id': '',
```

Comment:
I think we can remove these scrollbar/scrollbarbox items

Reply:
thanks  : )